I have implemented all the mentioned tasks in that exercise.

BilalLiaquat7 opened the pull request:
In this PR: I have implemented all the queries with their screenshots in folder named pull request 2
Folder name pull request 2 which contains 4 screenshots.
queries file updated
schema file updated
data file updated

This  image for first transaction
![image](https://github.com/BilalLiaquat7/vet_clinic_database/assets/21335436/c0f831d6-c795-4825-9f78-bdab626ddf1f)

This image for second transaction.
![photo 2](https://github.com/BilalLiaquat7/vet_clinic_database/assets/21335436/31cace11-fad1-4c53-a2a5-753924ae4d79)

This image for 3rd transaction
![photo 3](https://github.com/BilalLiaquat7/vet_clinic_database/assets/21335436/c9388c58-195c-4569-a379-92faf2510f8c)

This image for 4th transaction
![photo 4](https://github.com/BilalLiaquat7/vet_clinic_database/assets/21335436/dd40b1d7-c9c4-478c-94ed-4e6377a6b44f)

This image for 5th transaction.
![photo 5](https://github.com/BilalLiaquat7/vet_clinic_database/assets/21335436/7752fcff-9ada-4c55-accc-b8aee2c2fa69)

BilalLiaquat7 committed:
added new column and data added successfully into table

BilalLiaquat7 committed:
schema updated

BilalLiaquat7 committed:
all queries added along with screenshots

Strangeal commented on queries.sql:
```diff
@@ -29,4 +29,82 @@ WHERE
+
+ROLLBACK;
+
+SELECT * from animals
```
- [ ] Great work on your query to update the `species` column in the animal table! 🐾👏 However, I'd like to highlight a specific requirement from the project guidelines: it explicitly states that after updating the `species` column to `unspecified`, verify that the change was made successfully. Then roll back the change and `verify` that the `species` columns went back to the state before the transaction. 🔄

  - Upon reviewing your query, I noticed that you successfully updated the `species` column to `unspecified` within a transaction. However, it appears that you missed one of the verification steps outlined in the project guidelines. These verification steps are crucial to ensuring the integrity of the data and the correctness of your query.

    **To meet the project requirements, please consider adding the following verification step to your query:**

     1. After the update, verify that the change was made successfully. You can do this by executing a SELECT statement to confirm that the `species` column now contains the value `unspecified` just like you did on line `42` after the roll back. 🧐

Adding these verification steps will not only ensure that your query meets the project requirements but also provide a robust and reliable way to confirm the correctness of your data manipulation.

Happy coding! 😊🚀🍀

Strangeal commented on queries.sql:
```diff
@@ -29,4 +29,82 @@ WHERE
+-- Verify that changes persist after commit.
+-- Take a screenshot of the results of your actions.
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name like '%mon%';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * from animals;
+
+COMMIT;
```
- [ ] Great work on crafting a transactional query to update the `species` column in the `animal` table! 🎉 However, as I highlighted in my previous comment regarding the significance of incorporating verification steps within transactional queries, it seems like one verification step is still missing. This step should be implemented after the `COMMIT` to ensure that the change persists successfully following the commit. 👍

   - Adding this verification step will help guarantee the integrity of your data and provide an extra layer of confidence in the transaction's success. 

Happy coding! 😊👩‍💻🚀

Strangeal commented on pull request 2/picture 2.png:
- [ ] Great work on including screenshots in your pull request to illustrate the results of your queries! 👍 However, for better adherence to the project requirements, it would be more appropriate to follow the instructions and include the screenshot within the pull request summary. This will ensure that all project-related content is easily accessible and well-organized for reviewers and team members. 📸👀

Happy coding! 🚀😊

Strangeal commented on queries.sql:
- [ ] Great job so far on your effort! 🚀 However, I'd like to bring to your attention that there are some aspects of the project requirements that still need attention. In the project guidelines, you were provided with a specific list of cases for which you were asked to write queries and include screenshots of the query results in your pull request summary which you haven't added yet. 📊📸

Here's a screenshot of the cases you need to write queries for, just as a reminder:
<img width="809" alt="Screenshot 2023-09-27 at 14 57 01" src="https://github.com/BilalLiaquat7/vet_clinic_database/assets/107192076/2d714c4b-2af4-4cc6-987f-6c89e7d3cc2d">

   - Including these query results will ensure that you've covered all the required cases. It's an essential step in aligning your work with the project's objectives. 😊

Happy coding! 😊🚀🍀

Strangeal reviewed (changes requested):
Hi @BilalLiaquat7, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExOGYyYWNkMjRkY2JjYjg1MjY1M2NjYTNkNWNiNTQwN2EyZGZmNWM5YSZlcD12MV9pbnRlcm5hbF9naWZzX2dpZklkJmN0PWc/m3864rBwwBTKMAbpn3/giphy.gif)

## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Well-structured project ✅ 

## Required Changes ♻️
- [ ] Great job on documenting your project so thoroughly! 📚 However, I noticed that the link to your `MIT` license is currently broken and leads to a `404` error when clicked. It's important to fix this issue to ensure that your users have access to the necessary licensing information without any hindrance. Please take a moment to update the link so that your project remains compliant and user-friendly. 👍
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BilalLiaquat7 committed:
made changes as per reviewr asked

Strangeal commented on queries.sql:
```diff
@@ -107,4 +111,24 @@ SELECT * from animals;
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY (neutered);
+
+SELECT min(weight_kg), max(weight_kg) from animals;
+
```
- [ ] Great job so far in your implementation. However, I would like you to take a second look at this query as it doesn't return the correct results. The query is to return the min and max weight of `each` type of animal. In your animals table you have only two types of animals `Pokemon` and `Digimon` and you're to return the min and max weight of each of these types. Consider rewriting your query to return the written results.

    - Also, as I mentioned in my preview comments. For better adherence to the project requirements, it would be more appropriate to follow the instructions and include the `screenshot` within the `pull request summary`. This will ensure that all project-related content is easily accessible and well-organized for reviewers and team members. 📸👀

Happy coding! 🚀😊

Strangeal commented on queries.sql:
```diff
@@ -107,4 +111,24 @@ SELECT * from animals;
+
+SELECT avg(weight_kg) from animals;
+
+SELECT neutered, count(*) as escape_attempts
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY (neutered);
```
- [ ] 👍 Great effort on writing your query! However, there's a small issue with the current query, and I'd like to help you make it even better. 

   - The goal of the query is to determine who escapes more frequently, either `neutered` or `not neutered` animals. To achieve this, you should modify the query to retrieve the results where `neutered` is `true` and `escape_attempts` is equal to `20`. 

Here's an example of a refined version of the query you can use to get the desired results:

```
SELECT neutered, SUM(escape_attempts) AS total_escapes
	FROM animals
	GROUP BY neutered
	ORDER BY total_escapes DESC
	LIMIT 1;
```

Happy coding! 😊👏

Strangeal reviewed (changes requested):
Hi @BilalLiaquat7, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![Alt Text](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)

 ## Highlights
-  Added query for all cases

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BilalLiaquat7 committed:
changings as per reviewer

addod19 reviewed (changes requested):
## Hi @BilalLiaquat7, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- [ ] Nice job addressing some of the comments from the previous review :rocket: 

 ## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we include the screenshot of the query results in the PR as mentioned in the previous review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BilalLiaquat7 commented:
@addod19 
Please check again there is a folder that contains all screenshots that is mentioned in the exercise. do ask via Zoom rather than posting a vague review here.
Unhappy with your review.

BilalLiaquat7 commented:
> ## Hi @BilalLiaquat7,
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights
> * [ ]  Nice job addressing some of the comments from the previous review 🚀
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> * [ ]  It would be better if we include the screenshot of the query results in the PR as mentioned in the previous review
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

"@addod19
Please check again there is a folder that contains all screenshots that is mentioned in the exercise. do ask via Zoom rather than posting a vague review here.
Unhappy with your review."

HAMMAS-SALEEM commented on queries.sql:
```diff
@@ -29,4 +29,109 @@ WHERE
+WHERE escape_attempts = 0;
+
+SELECT avg(weight_kg) from animals;
+
+SELECT neutered, SUM(escape_attempts) AS total_escapes
+	FROM animals
+	GROUP BY neutered
+	ORDER BY total_escapes DESC
+	LIMIT 1;
+
```
**[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] Great work so far but I guess you misunderstood the requirement because you are supposed to get that individual animal who escapes the most from the neutered and non-neutered animals 🤔 so you should use `AVG` or `MAX` instead of SUM because SUM is doing summation of all the escape attempts which we don't want 😨

HAMMAS-SALEEM commented on queries.sql:
```diff
@@ -29,4 +29,109 @@ WHERE
+Commit;
+
+SELECT * from animals
+
+-- ////Picture 4///
+
```
- [ ] Great work on making the screenshots but i guess you misunderstood the requirement here because the requirement is to add all the screenshots in the pull request descriptions means the screenshots themselves 🤔 So kindly add all the screenshots in the pull request description to make your work follow the project requirements 👍

HAMMAS-SALEEM reviewed (changes requested):
# Required Changes :recycle:
![Ukhv](https://user-images.githubusercontent.com/93039359/247195854-e274f284-2e3a-423b-8939-7e4efb638ee0.gif)

## Hey @BilalLiaquat7 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements
✔️ Great work on fixing the transactions 👍
✔️ Great work on having the screenshots in your repo 🔥

Good job on implementing some of the requested changes, There are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BilalLiaquat7 committed:
commit as per requested changes

mukhammadrizooff reviewed (approved):
Hi @BilalLiaquat7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

<img src="https://media4.giphy.com/media/sR9nilEznDoSlga667/giphy.gif"/> 

## Highlights
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
